src/overlay/Peer.cpp

```diff
@@ -861,83 +865,101 @@ getOutboundFloodTimer(Application& app, StellarMessage const& msg)
 }
 
 void
-Peer::maybeTrimQueue()
+Peer::maybeTrimQueue(MessageType lastEnqueuedMessageType)
 {
-    auto now = mApp.getClock().now();
-    auto shouldTrim = [&](QueuedOutboundMessage const& msg) -> bool {
-        if (msg.mMessage.type() == TRANSACTION)
-        {
-            // Trim transactions that sit in the queue for too long
-            return (now - msg.mTimeEmplaced) >
-                   Peer::OUTBOUND_TRANSACTION_DROP_TIMEOUT;
-        }
-        else if (msg.mMessage.type() == SCP_MESSAGE)
-        {
-            auto slotIndex = msg.mMessage.envelope().statement.slotIndex;
-            auto minSlotToRemember =
-                mApp.getHerder().getMinLedgerSeqToRemember();
-            if (slotIndex >= minSlotToRemember &&
-                mApp.getHerder().isLatestSCPMessage(msg.mMessage))
-            {
-                return false;
-            }
-            // Trim an SCP message if it's outside of range or we have a newer
-            // statement
-            return true;
-        }
-        else
-        {
-            return false;
-        }
-    };
-
     int dropped = 0;
-    auto it = mFlowControlledQueue.cbegin();
-    while (it != mFlowControlledQueue.cend())
+    if (lastEnqueuedMessageType == TRANSACTION)
     {
-        if (shouldTrim(*it))
+        auto& queue = mOutboundQueues[1];
+
+        while (queue.size() > Peer::OUTBOUND_TRANSACTION_DROP_LIMIT)
```

Comment:
Should the max queue size for transactions should just be related to the maximum rate of transactions that can be included in a ledger `mApp.getLedgerManager().getLastMaxTxSetSizeOps()`? This avoids adding this hardcoded value.